### DISCUSSION TITLE
Svg selector

### DISCUSSION THREAD

**Discussion by Davide-Guidetti** [clientIO/joint#2707] [Q&A] [answered]
### Introduction

Goodmorning,
I am currently working on a project using jointjs and I am making extensive use of selectors (@selector). However, I have encountered an issue where the SVG elements cannot be rendered in the browser because @selector is not a native SVG tag.
I would like to inquire if there are any alternative approaches to handling selectors that would allow me to visualize them in the browser and verify their functionality.
Any insights or suggestions you can provide would be greatly appreciated.
Thank you in advance for your assistance.
Davide

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
You are most likely not using the [util.svg](https://docs.jointjs.com/api/util/#svg) to parse your stringified XML markup to JSON markup.

The `@selector` must be used with the [tagged template](https://docs.jointjs.com/learn/features/shapes/markup/#json-markup-using-tagged-template).  Then it is rendered as `joint-selector` in the browser.

**Reply by Davide-Guidetti** [reply]
Good morning, Roman,
Perhaps I didn't express myself clearly.
I am using util.svg and I can correctly display the element within the jointjs component. The difficulty arises if I want to open the SVG outside of jointjs to check its structure. In that case, the presence of @selector inevitably causes display issues.
I was wondering if there was an alternative to correctly display the SVG both inside the jointjs component and "normally."
Thanks in advance,
Davide

**Reply by kumilingus** [reply, marked as answer]
You can call it `data-selector` in the SVG.
Then you load the SVG as a string,  replace `data-selector` with `@selector` and pass it to `util.svg`.
